### DISCUSSION TITLE
Morph Update after velocity computation

### DISCUSSION THREAD

**Discussion by neinecke** [esa/SpOC2#11] [Morphing Rovers Q&A] [answered]
It seems that the morphing update is done after computing the velocities, i.e. the new morph is only effective starting from the next time step. Is that really the intended behavior of the rover?

**Comment by DreamTlaloc** [comment, marked as answer]
Hi Neinecke,

You are correct that the the order of operations makes new forms activate in the next time step, this can be interpreted as the time it takes to morph into a new form once the controller decides to morph. In general, the morphing constraints exist to prevent exploitation of the morphing concept and make the results more 'realistic' without overly constraining the strategies that competitors can pursue, i.e. there is a cooldown to prevent constant switching, but there is no score penalty from morphing.

Thanks,
Amy
